src/main/java/com/minecolonies/coremod/placementhandlers/JigsawPlacementHandler.java

```diff
@@ -0,0 +1,125 @@
+package com.minecolonies.coremod.placementhandlers;
+
+import com.google.common.collect.ImmutableList;
+import com.ldtteam.structurize.placement.handlers.placement.IPlacementHandler;
+import com.ldtteam.structurize.util.BlockUtils;
+import com.ldtteam.structurize.util.PlacementSettings;
+import com.minecolonies.api.util.Log;
+import com.minecolonies.api.util.WorldUtil;
+import com.mojang.brigadier.StringReader;
+import com.mojang.brigadier.exceptions.CommandSyntaxException;
+import net.minecraft.commands.arguments.blocks.BlockStateParser;
+import net.minecraft.core.BlockPos;
+import net.minecraft.nbt.CompoundTag;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.level.Level;
+import net.minecraft.world.level.block.Blocks;
+import net.minecraft.world.level.block.JigsawBlock;
+import net.minecraft.world.level.block.StructureBlock;
+import net.minecraft.world.level.block.StructureVoidBlock;
+import net.minecraft.world.level.block.state.BlockState;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.Collections;
+import java.util.List;
+
+import static com.ldtteam.structurize.placement.handlers.placement.PlacementHandlers.handleTileEntityPlacement;
+
+/**
+ * Handler for some specific blocks that we want only to paste/cost in the complete mode.
+ */
+public class JigsawPlacementHandler implements IPlacementHandler
+{
+    @Override
+    public boolean canHandle(@NotNull final Level world, @NotNull final BlockPos pos, @NotNull final BlockState blockState)
+    {
+        return blockState.getBlock() instanceof JigsawBlock;
+    }
+
+    @Override
+    public ActionProcessingResult handle(
+      @NotNull final Level world,
+      @NotNull final BlockPos pos,
+      @NotNull final BlockState blockState,
+      @Nullable final CompoundTag tileEntityData,
+      final boolean complete,
+      final BlockPos centerPos,
+      final PlacementSettings settings)
+    {
+        if (complete)
+        {
+            WorldUtil.setBlockState(world, pos, blockState, com.ldtteam.structurize.api.util.constant.Constants.UPDATE_FLAG);
+            if (tileEntityData != null)
+            {
+                try
+                {
+                    handleTileEntityPlacement(tileEntityData, world, pos, settings);
+                    blockState.getBlock().setPlacedBy(world, pos, blockState, null, BlockUtils.getItemStackFromBlockState(blockState));
+                }
+                catch (final Exception ex)
+                {
+                    Log.getLogger().warn("Unable to place TileEntity");
+                }
+            }
+            return ActionProcessingResult.SUCCESS;
+        }
+
+        if (tileEntityData != null && tileEntityData.contains("final_state"))
+        {
+            final String stateString = tileEntityData.getString("final_state");
+            BlockStateParser blockstateparser = new BlockStateParser(new StringReader(stateString), false);
```

Comment:
It might be nice if you could invoke `JigsawReplacementProcessor.processBlock` instead of hard-coding this data/behaviour.  Just in case Mojang decide to change the storage or something in the future.  It's basically the same code as this.

Reply:
Didn't want to do it because of potential side-effects. If they change the storage, it will just stop working for us and we'll directly notice it, but if they change sth to have side-effects it will be hard to find.